src/test/app/LedgerLoad_test.cpp

```diff
@@ -63,7 +63,7 @@ class LedgerLoad_test : public beast::unit_test::suite
         retval.ledgerFile = td.file("ledgerdata.json");
 
         Env env{*this};
-        Account prev;
+        Account prev("previous_account");
```

Comment:
About 10 lines below, `prev` is used in `env.trust()` and `env(pay(...))` commands. I don't think `prev` can be used as a recipient account if its an unseated optional

Reply:
Yes, but those transactions are skipped on the first run through the loop. `prev` is assigned at the end of the loop. So it's guaranteed that `prev` will have a value when it is dereferenced.

```
        std::optional<Account> prev;

        for (auto i = 0; i < 20; ++i)
        {
            Account acct{"A" + std::to_string(i)};
            env.fund(XRP(10000), acct);
            env.close();
            if (i > 0 && BEAST_EXPECT(prev))
            {
                env.trust(acct["USD"](1000), *prev);
                env(pay(acct, *prev, acct["USD"](5)));
            }
            env(offer(acct, XRP(100), acct["USD"](1)));
            env.close();
            prev.emplace(std::move(acct));
        }
```

Another, simpler option is instead of initializing it with a dummy value that's going to get thrown away, initialize it with a value you know it will have. At least then you know it will be memoized.
```
Account prev("A19");
```

Reply:
thanks Ed, this is very thoughtful!

Reply:
Hmm, that's an interesting thought. Surprisingly, the `operator=` code generates 10-20 lines more assembly than the `emplace` code. I'm guessing there are other aspects of performance to consider. https://godbolt.org/z/hW1a4GroP